File: src/main/java/com/adiwal/auth/repository/UserRepository.java
package com.adiwal.auth.repository;

import java.util.Optional;

import org.springframework.data.mongodb.repository.MongoRepository;

import com.adiwal.auth.models.User;

public interface UserRepository extends MongoRepository<User, String> {
  Optional<User> findByUsername(String username);

  Boolean existsByUsername(String username);

  Boolean existsByEmail(String email);
}

File: src/main/kotlin/com/adiwal/auth/repository/UserRepository.kt
package com.adiwal.auth.repository

import com.adiwal.auth.domain.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import org.springframework.transaction.annotation.Propagation
import org.springframework.transaction.annotation.Transactional
import java.util.*

@Repository
@Transactional(propagation = Propagation.MANDATORY)
interface UserRepository: JpaRepository<User, Long> {
    fun findByUsername(username: String): Optional<User>
}